# manifest: PT-025
# base: 868d8d9216da361c666519652115e23688db5211
# mechanism: capability
# expiry: TBD
# tests: tests/pytorch/test_numerics.py, tests/pytorch/mxfp4, tests/pytorch/test_fusible_ops.py
# owner: TBD
# seeded: 2026-08-30 whole-file diff of the fork vs upstream; P5 shrinks/splits this (0 feature sub-entries)
--- a/transformer_engine/pytorch/tensor/float8_blockwise_tensor.py
+++ b/transformer_engine/pytorch/tensor/float8_blockwise_tensor.py
@@ -1,3 +1,5 @@
+# This file was modified for portability to AMDGPU
+# Copyright (c) 2026, Advanced Micro Devices, Inc. All rights reserved.
 # Copyright (c) 2022-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 #
 # See LICENSE for license information.
@@ -22,6 +24,8 @@
 from ..constants import DType
 from ..utils import devices_match, round_up_to_nearest_multiple
 
+from torch.utils.cpp_extension import IS_HIP_EXTENSION
+
 aten = torch.ops.aten
 
 
@@ -159,19 +163,27 @@
             scale_dim0 = (dim0 + self.block_len - 1) // self.block_len
             scale_dim1 = (dim1 + self.block_len - 1) // self.block_len
             if columnwise:
+                if IS_HIP_EXTENSION:
+                    return (scale_dim1, scale_dim0)
                 return (scale_dim1, round_up_to_nearest_multiple(scale_dim0, 4))
+            if IS_HIP_EXTENSION:
+                return (scale_dim0, scale_dim1)
             return (scale_dim0, round_up_to_nearest_multiple(scale_dim1, 4))
 
         # 1x128 block scaling
         if columnwise:
-            return (
-                (dim0 + self.block_len - 1) // self.block_len,
-                round_up_to_nearest_multiple(dim1, 4),
-            )
-        return (
-            (dim1 + self.block_len - 1) // self.block_len,
-            round_up_to_nearest_multiple(dim0, 4),
-        )
+            outer = (dim0 + self.block_len - 1) // self.block_len
+            if IS_HIP_EXTENSION:
+                inner = dim1
+            else:
+                inner = round_up_to_nearest_multiple(dim1, 4)
+            return (outer, inner)
+        outer = (dim1 + self.block_len - 1) // self.block_len
+        if IS_HIP_EXTENSION:
+            inner = dim0
+        else:
+            inner = round_up_to_nearest_multiple(dim0, 4)
+        return (outer, inner)
 
     def get_columnwise_shape(self, shape: Iterable[int]) -> Tuple[int, ...]:
         """Column-wise data shape
